client/components/CheckoutButton.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import {connect} from 'react-redux'
+import {createCheckoutSession} from '../store/checkout'
+// import {Routes, Route, useNavigate} from 'react-router-dom'
+
+export class CheckoutButton extends React.Component {
+  render() {
+    // const navigateToCheckout = useNavigate('/checkout')
+    return (
+      <div>
+        {/* <button
+          type="button"
+          onClick={() => this.props.createCheckoutSession()}
+        >
+          Checkout
+        </button> */}
+        <form action="/create-checkout-session" method="POST">
```

Comment:
Clicking the checkout button should really be dispatching a thunk that makes the checkout session POST from in code. This will also force a page-refresh which we want to avoid in our single page apps. Finally, you're also going to need to send your user token with that request to validate who you are (a user should only be able to checkout their own order)